numpy/core/_dtype.py

```diff
@@ -348,7 +350,9 @@ def _name_get(dtype):
         # user dtypes don't promise to do anything special
         return dtype.type.__name__
 
-    if issubclass(dtype.type, np.void):
+    if dtype.kind == '\x00':
```

Comment:
We really should improve thi.  Right now I have `type(dtype)._parametric` and `_abstract`.  I am thinking it might make sense to make these public and  (maybe without the underscore).
Could also add it to the dtype instances.  I think we can consider these attributes stable API really (maybe even already with that underscore).

Reply:
Yeah, I think having a concrete and obvious way to signal that this is a new-style dtype would be great. I don't have a strong opinion about how that should be spelled.

Reply:
> I don't have a strong opinion about how that should be spelled.

If not you, who then? :).  More seriously though, we have to make these public in some form I think.  And if we are unsure, a safe bet is to just keep it as is (with an underscore), but define it as stable API (and maybe document it).

Reply:
Returning 3 there seems OK, the only danger is if someone out there did `dtype.isbuiltin != 2` or something crazy.  But that seems unlikely?!  `dtype.iscustom` is maybe not so bad (I would just ignore the "legacy" part and assume legacy dtypes will die off eventually?)

I do think we need this information also on the type probably in the long term.  One other way to do this is always `isinstance/issubclass(..., np.UserDType)`.  That would probably make Marten very happy I am sure ;).  I do not see that such APIs as mutual exclusive, though.